runtime/src/test/scala/akka/grpc/internal/NettyClientUtilsSpec.scala

```diff
@@ -26,12 +26,13 @@ class NettyClientUtilsSpec extends AnyWordSpec with Matchers with ScalaFutures w
   "The Netty client-utilities" should {
     implicit val ec = system.dispatcher
 
-    "fail to create a channel when service discovery times out" in {
-      val settings = GrpcClientSettings.usingServiceDiscovery("testService")
-
-      val channel = NettyClientUtils.createChannel(settings, Logging(system, this.getClass))
-      channel.failed.futureValue shouldBe a[AskTimeoutException]
-    }
+//    The channel can now retry service discovery as needed itself,
+//    I guess we should test that instead?
+//    "fail to create a channel when service discovery times out" in {
+//      val settings = GrpcClientSettings.usingServiceDiscovery("testService")
+//
+//      val channel = NettyClientUtils.createChannel(settings)
+//    }
   }
```

Comment:
With the changes in `NettyClientUtils` the [invocation](https://github.com/akka/akka-grpc/pull/811/files#diff-977da58f66c832d603f6ad9ac553d99dR71 to `val channel = builder.build()` is not longer within a `try{}catch` (or a `future.map{}` as it used to be). Therefore an invocation to `createChannel()` will either return a successful `Future` or throw a runtime exception but never a `Future.failed`?

Reply:
I didn't mean for that code to ever throw runtime exceptions. I kept the `Future` in the API for now to not change too much at once.